opal/corelib/string.rb

```diff
@@ -8,7 +8,18 @@
 class ::String < `String`
```

Comment:
Can the changes on String/Encoding be separated to another PR?

Reply:
I added another section to the top comment here "Desired Extractions", please add everything you want in a separate PR there or mention in a comment and i will add and i will then attend to them once everything is working.